src/gui_common/CustomDropDown.cs

```diff
@@ -175,10 +175,16 @@ public void CreateElements()
                 {
                     if (item.Checkable)
                     {
-                        Popup.AddIconCheckItem(item.Icon, item.Text, id);
+                        // Use a pre-tinted icon so the species color is applied only to the custom icon
+                        // while the radio/checkbox indicator (selection control) keeps theme colors
+                        var tintedIcon = item.PreTintedIcon ??= CreateTintedIcon(item.Icon, item.Color);
+
+                        Popup.AddIconCheckItem(tintedIcon, item.Text, id);
                         var index = Popup.GetItemIndex(id);
                         Popup.SetItemIconMaxWidth(index, iconMaxWidth);
-                        Popup.SetItemIconModulate(index, item.Color);
```

Comment:
Upon reading that this SetItemIconModulate is from Godot, I think this might actually be an engine bug that should be reported that Godot 4 now applies icon modulation to the radiobutton as well rather than the icon only.